agent/local/state.go

```diff
@@ -378,6 +380,10 @@ func (l *State) setServiceStateLocked(s *ServiceState) {
 
 	if hasOld && old.WatchCh != nil {
 		close(old.WatchCh)
+	} else {
+		// The status of an alias check is updated if the alias service is added/removed
+		// Only try notify alias checks if service didn't already exist (!hasOld)
```

Comment:
Note that the else block is triggered for `!hasOld || old.WatchCh == nil` not just `!hasOld` so you might want an additional guard here, or switch this to:
```
if hasOld && old.WatchCh != nil {
...
} else if !hasOld {
...
}
```
instead

Reply:
Yep, switched to just check for `!hasOld`